File: week1/pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Prompts user for number
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Prints pyramid
    int space = height - 1;
    int hash = 1;
    int width = 1;

    while (hash <= height)
    {
        // SPACES
        while (space > 0)
        {
            printf(" ");
            space--;
        }

        // LEFT HASHES
        width = hash;
        while (hash > 0)
        {
            printf("#");
            hash--;
        }
        hash = width;

        // GAP
        printf("  ");

        //RIGHT HASHES
        while (hash > 0)
        {
            printf("#");
            hash--;
        }

        // NEW LINE
        printf("\n");
        hash = width + 1;
        space = height - hash;
    }
}

File: week6/pset6/mario/mario.py
from cs50 import get_int

height = 0

while height < 1 or height > 8:
    height = get_int("Height: ")
    
space = height - 1
hash_number = 1
width = 1

while hash_number <= height:
    
    # Spaces
    while space > 0:
        print(" ", end="")
        space -= 1
    
    # Left Hashes
    width = hash_number
    while hash_number > 0:
        print("#", end="")
        hash_number -= 1
    hash_number = width
    
    # Gaps
    print("  ", end="")
    
    # Right Hashes
    while hash_number > 0:
        print("#", end="")
        hash_number -= 1
        
    # New Line
    print("")
    hash_number = width + 1
    space = height - hash_number
    